src/linker/Linker/LinkContext.cs

```diff
@@ -465,9 +468,17 @@ public bool IsOptimizationEnabled (CodeOptimizations optimization, AssemblyDefin
 
 		public void LogMessage (MessageContainer message)
 		{
-			if (LogMessages)
+			if (LogMessages && message != MessageContainer.Empty)
 				Logger?.LogMessage (message);
 		}
+
+		public bool IsWarningSuppressed (int warningCode, MessageOrigin origin)
+		{
+			if (Suppressions == null)
+				return false;
+
+			return Suppressions.IsSuppressed ("IL" + warningCode, origin, out _);
```

Comment:
Can we avoid the "IL" + warningCode. This means allocation every time we call this method. Seems unnecessary. If we only support warnings of shape IL####, then we can probably parse the suppression attributes directly to the warning code.